packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -176,133 +178,162 @@ class _ZoomPageTransition extends StatefulWidget {
-      Tween<double>(begin: 0.85, end: 1.00)
-        .chain(_ZoomPageTransition._scaleCurveSequence));
-
-    final Animation<double> _forwardStartScreenScaleTransition = widget.secondaryAnimation.drive(
-      Tween<double>(begin: 1.00, end: 1.05)
-        .chain(_ZoomPageTransition._scaleCurveSequence));
-
-    final Animation<double> _forwardEndScreenFadeTransition = widget.animation.drive(
-      Tween<double>(begin: 0.0, end: 1.00)
-        .chain(CurveTween(curve: const Interval(0.125, 0.250))));
-
-    final Animation<double> _reverseEndScreenScaleTransition = widget.secondaryAnimation.drive(
-      Tween<double>(begin: 1.00, end: 1.10)
-        .chain(_ZoomPageTransition._flippedScaleCurveSequence));
-
-    final Animation<double> _reverseStartScreenScaleTransition = widget.animation.drive(
-      Tween<double>(begin: 0.9, end: 1.0)
-        .chain(_ZoomPageTransition._flippedScaleCurveSequence));
-
-    final Animation<double> _reverseStartScreenFadeTransition = widget.animation.drive(
-      Tween<double>(begin: 0.0, end: 1.00)
-        .chain(CurveTween(curve: const Interval(1 - 0.2075, 1 - 0.0825))));
-
-    return AnimatedBuilder(
-      animation: widget.animation,
-      builder: (BuildContext context, Widget child) {
-        if (widget.animation.status == AnimationStatus.forward || _transitionWasInterrupted) {
-          return Container(
-            color: Colors.black.withOpacity(_forwardScrimOpacityAnimation.value),
-            child: FadeTransition(
-              opacity: _forwardEndScreenFadeTransition,
-              child: ScaleTransition(
-                scale: _forwardEndScreenScaleTransition,
-                child: child,
-              ),
-            ),
-          );
-        } else if (widget.animation.status == AnimationStatus.reverse) {
-          return ScaleTransition(
-            scale: _reverseStartScreenScaleTransition,
-            child: FadeTransition(
-              opacity: _reverseStartScreenFadeTransition,
-              child: child,
-            ),
-          );
-        }
-        return child;
-      },
-      child: AnimatedBuilder(
-        animation: widget.secondaryAnimation,
-        builder: (BuildContext context, Widget child) {
-          if (widget.secondaryAnimation.status == AnimationStatus.forward || _transitionWasInterrupted) {
-            return ScaleTransition(
-              scale: _forwardStartScreenScaleTransition,
-              child: child,
-            );
-          } else if (widget.secondaryAnimation.status == AnimationStatus.reverse) {
-            return ScaleTransition(
-              scale: _reverseEndScreenScaleTransition,
-              child: child,
-            );
-          }
-          return child;
-        },
-        child: widget.child,
+    // Only change transitions when the animation was not dismissed.
+    final bool isDismissed = animation.status == AnimationStatus.dismissed;
+    return ScaleTransition(
+      scale: !isDismissed
```

Comment:
nit: can you turn this around for easier readability: isDismissed ? kAlwaysCompleteAnimation : foo.

Reply:
here and elsewhere